crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/acir_variable.rs

```diff
@@ -443,7 +450,7 @@ impl AcirContext {
     pub(crate) fn black_box_function(
         &mut self,
         name: BlackBoxFunc,
-        inputs: Vec<AcirVar>,
+        inputs: Vec<BlackboxArg>,
```

Comment:
why can't the function that is calling this deconstruct the arrays into AcirVar?

Reply:
Perhaps if a black box function takes in multiple arguments, it becomes non-trivial

Reply:
ie `std::sha256([x], [y])` Would compile to `vec![AcirVar(0), AcirVar(1)]`

Reply:
I think this is the case for recursion (kinda) -- we can move it to acir_gen/mod and reconsider this once everything has settled. Can you create an issue for this?

Reply:
Moved to acir_gen/mod